woocommerce/woocommerce-ios
# Store onboarding: Handle store name setup

Part of #10463 


## Description

This PR adds UI and handles saving store name from the onboarding task for free trial sites with the default name.

Changes:
- Add new view and view model for setting up store name.
- Present this new view from the `StoreOnboardingCoordinator`.
- Update `SiteStore` to return the updated site upon store name update success. This is necessary to update the selected store and display the updated store name on the My Store screen immediately.

## Testing instructions

- Enable the feature flag `optimizeProfilerQuestions` and build the app.
- Log in and switch to the Menu tab.
- Select the store picker and tap Add a store > Create a new store.
- Follow the steps to create a free trial store.
- When the store creation completes, you should be navigated to the My Store screen.
- Notice a new onboarding task "Name your store" on the onboarding task list.
- Tap the onboarding task to update store name. Notice a new screen is displayed.
- The Save button should be disabled when the store name is not updated or is empty.
- Disable your internet connection.
- Change the store name to a new name and tap Save. The view should display the activity indicator when saving is in progress.
- An error message should be displayed below the text field when the request times out.
- Turn on your connection and try saving the store name again.
- When saving completes, the view should be dismissed, and the updated name should be displayed on the My Store screen.
- The store name update onboarding task should be marked as complete.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/d6ea31c8-756c-4212-bf1b-2e596873b222



---
